packages/lib/modules/pool/PoolDetail/PoolStats/PoolSnapshot/PoolSnapshotValues.tsx

```diff
@@ -16,13 +17,30 @@ type PoolStatsValues = {
 }
 
 export function PoolSnapshotValues() {
-  const { pool, chain, tvl } = usePool()
+  const { chain, pool, tvl } = usePool()
   const { toCurrency } = useCurrency()
 
   const MemoizedMainAprTooltip = memo(MainAprTooltip)
 
   const { tokens, weeklyRewards } = useGetPoolRewards(pool)
 
+  const tokenBalances = useMemo(() => {
```

Comment:
This logic is not super hard but also not super straight forward. I would push this `useMemo` inside `useGetPoolRewards` hook that would expose a new `weeklyRewardsByToken` result.

And I would add tests to the already existing `useGetPoolRewards.spec.tsx`